plugins/Phaser/PhaserControls.cpp

```diff
@@ -0,0 +1,143 @@
+/*
+ * PhaserControls.cpp
+ *
+ * Copyright (c) 2019 Lost Robot <r94231@gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+
+#include <QDomElement>
+
+#include "PhaserControls.h"
+#include "PhaserEffect.h"
+
+#include "Engine.h"
+#include "GuiApplication.h"
+#include "MainWindow.h"
+#include "Song.h"
+
+
+constexpr int PHA_DOT_SLIDER_LENGTH = 338;
+constexpr float PHA_MIN_FREQ = 20;
+constexpr float PHA_MAX_FREQ = 20000;
+
+
+PhaserControls::PhaserControls(PhaserEffect* effect) :
+	EffectControls(effect),
+	m_effect(effect),
+	m_cutoffModel(640.0f, 20.0f, 20000.0f, 0.01f, this, tr("Cutoff")),
+	m_resonanceModel(0.05f, 0.05f, 2.0f, 0.001f, this, tr("Resonance")),
+	m_feedbackModel(0.0f, -100.0f, 100.0f, 0.01f, this, tr("Feedback")),
+	m_orderModel(8, 0, 32, this, tr("Order")),
+	m_delayModel(0.0f, 0.0f, 20.0f, 0.001f, this, tr("Delay")),
+	m_rateModel(9.0f, 0.01f, 60.0f, 0.001f, 60000.0, this, tr("Rate")),
+	m_amountModel(4.5f, 0.f, 5.0f, 0.01f, this, tr("Amount")),
+	m_enableLFOModel(true , this, tr("Enable LFO")),
+	m_phaseModel(180.0f, 0.f, 360.0f, 0.01f, this, tr("Phase")),
+	m_inFollowModel(0.f, -30.f, 30.0f, 0.01f, this, tr("Input Follow")),
+	m_attackModel(200.0f, 0.0f, 2000.0f, 1.0f, this, tr("Attack")),
+	m_releaseModel(500.0f, 0.0f, 2000.0f, 1.0f, this, tr("Release")),
```

Comment:
m_attackModel and m_releaseModel both have a minimum of 0.0f which leads to division with 0 in calcAttack() and calcRelease() respective. Choose a minimum value here or guard against 0 in mentioned functions. I suggest a minimum value here.

Reply:
Edit. The AppImage doesn't see any problems because of this but we have built-in debug stuff that catches floating point errors. If we don't appease the machine chaos will ensue.